pom.xml

```diff
@@ -602,6 +610,20 @@
         </profile>
         <profile>
             <id>kafka-0.8-dev</id>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-compiler-plugin</artifactId>
+                        <version>2.0.2</version>
+                        <configuration>
+                            <excludes>
+                                <exclude>com/pinterest/secor/timestamp/Kafka10MessageTimestamp.java</exclude>
```

Comment:
The only problem with that approach is KafkaMessageTimestampFactory.java, because it will call either Kafka10MessageTimestamp.java or Kafka8MessageTimestamp.java depending on the config, and since one of the class will not be available for compile(because we have excluded it), so `cannot find symbol` error starts coming during compile. Thats why I found this workaround.